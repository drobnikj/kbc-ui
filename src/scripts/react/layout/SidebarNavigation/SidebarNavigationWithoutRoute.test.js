import React from 'react';
import { mount } from 'enzyme';

import SidebarNavigation from '../SidebarNavigation';

SidebarNavigation.contextTypes = {
  router: React.PropTypes.object.isRequired
};

jest.mock('../../../stores/ApplicationStore', () => {
  return {
    hasCurrentProjectFeature: () => {
      return false;
    },
    getProjectPageUrl: (pageId) => {
      return '/' + pageId;
    }
  };
});

jest.mock('../../../stores/RoutesStore', () => {
  return {
    hasRoute: () => {
      return false;
    }
  };
});

/* eslint-disable react/prop-types */
jest.mock('react-router/lib/components/Link', () => ({ to, children }) => {
  return (
    <a href={to}>{children}</a>
  );
});

const routerContext = {
  router: {
    isActive: function(id) {
      return id === 'extractors';
    }
  }
};

describe('<SidebarNavigation />', function() {
  it('should render standard navigation, without route', function() {
    matchSnapshot(
      mount(<SidebarNavigation />, { context: routerContext })
    );
  });
});
