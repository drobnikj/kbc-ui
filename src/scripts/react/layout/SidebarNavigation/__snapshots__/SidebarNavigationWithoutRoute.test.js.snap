exports[`<SidebarNavigation /> should render standard navigation, without route 1`] = `
<SidebarNavigation>
  <ul
    className="kbc-nav-sidebar nav nav-sidebar">
    <li
      className="">
      <a
        href="/home">
        <span
          className="kbc-icon-overview" />
        <span>
          Overview
        </span>
      </a>
    </li>
    <li
      className="active">
      <a
        href="/extractors">
        <span
          className="kbc-icon-extractors" />
        <span>
          Extractors
        </span>
      </a>
    </li>
    <li
      className="">
      <a
        href="/transformations">
        <span
          className="kbc-icon-transformations" />
        <span>
          Transformations
        </span>
      </a>
    </li>
    <li
      className="">
      <a
        href="/writers">
        <span
          className="kbc-icon-writers" />
        <span>
          Writers
        </span>
      </a>
    </li>
    <li
      className="">
      <a
        href="/orchestrations">
        <span
          className="kbc-icon-orchestrations" />
        <span>
          Orchestrations
        </span>
      </a>
    </li>
    <li
      className="">
      <a
        href="/storage">
        <span
          className="kbc-icon-storage" />
        <span>
          Storage
        </span>
      </a>
    </li>
    <li
      className="">
      <a
        href="/jobs">
        <span
          className="kbc-icon-jobs" />
        <span>
          Jobs
        </span>
      </a>
    </li>
    <li
      className="">
      <a
        href="/applications">
        <span
          className="kbc-icon-applications" />
        <span>
          Applications
        </span>
      </a>
    </li>
  </ul>
</SidebarNavigation>
`;
